test/junit/scala/reflect/internal/TypesTest.scala

```diff
@@ -61,6 +66,20 @@ class TypesTest {
     }
   }
 
+  @Test
+  def testNilModuleUnification(): Unit = {
+    import rootMirror.RootClass
+    val nil1 = singleType(ThisType(RootClass), typeOf[scala.`package`.type].member(TermName("Nil")))
+    val nil2 = typeOf[scala.collection.immutable.Nil.type]
```

Comment:
this is a `UniqueSingleType` (at least in the repl) not a `ModuleTypeRef`, so probably this test was passing already before your fix?

Reply:
Ah, yeah, I'd noticed that and then forgotten about it.  Good catch - fixed.